ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/KsqlRestApplication.java

```diff
@@ -262,7 +262,8 @@ protected void registerWebSocketEndpoints(final ServerContainer container) {
                       JsonMapper.INSTANCE.mapper,
                       statementParser,
                       ksqlEngine,
-                      exec
+                      exec,
+                      versionCheckerAgent.getLastRequestTime()
```

Comment:
This breaks encapsulation - you're sharing the AtomicLong between instances - better to pass an interface e.g. `Runnable onRequest` which these things call on a request.  Then have a single place, in the version checker agent, that all these `onRequest` callbacks are wired into that updates some internal state.  

This would be more decoupled, no shared state, and only a single place in the code that needs to dp the `lastRequestTime.set(timeUtils.nowInUnixTime());` bit. It also simplifies the testing, as only one thing needs to test its updating the time correctly, while all the others just need to check the event is getting invoked.

e.g. swap `getLastRequestTime` for:

```
 void updateLastRequestTime() {
    lastRequestTime.set(timeUtils.nowInUnixTime());  
 }
```

And then pass down to these using `versionCheckerAgent::updateLastRequestTime`

Reply:
I'll update the implementation as you suggested.